lib/Jsrt/Jsrt.cpp

```diff
@@ -2791,44 +2791,65 @@ CHAKRA_API JsConstructObject(_In_ JsValueRef function, _In_reads_(cargs) JsValue
     });
 }
 
-CHAKRA_API JsCreateFunction(_In_ JsNativeFunction nativeFunction, _In_opt_ void *callbackState, _Out_ JsValueRef *function)
+#ifndef _CHAKRACOREBUILD
+struct JsNativeFunctionInfo
 {
-    return ContextAPIWrapper<JSRT_MAYBE_TRUE>([&] (Js::ScriptContext *scriptContext, TTDRecorder& _actionEntryPopper) -> JsErrorCode {
-        PERFORM_JSRT_TTD_RECORD_ACTION(scriptContext, RecordJsRTAllocateFunction, false, nullptr);
-
-        PARAM_NOT_NULL(nativeFunction);
-        PARAM_NOT_NULL(function);
-        *function = nullptr;
+    JsValueRef thisArg;
+    JsValueRef newTargetArg;
+    bool isConstructCall;
+};
 
-        Js::JavascriptExternalFunction *externalFunction = scriptContext->GetLibrary()->CreateStdCallExternalFunction((Js::StdCallJavascriptMethod)nativeFunction, 0, callbackState);
-        *function = (JsValueRef)externalFunction;
+typedef _Ret_maybenull_ JsValueRef(CHAKRA_CALLBACK * JsEnhancedNativeFunction)(_In_ JsValueRef callee, _In_ JsValueRef *arguments, _In_ unsigned short argumentCount, _In_ JsNativeFunctionInfo *info, _In_opt_ void *callbackState);
+#endif
 
-        PERFORM_JSRT_TTD_RECORD_ACTION_RESULT(scriptContext, function);
+struct JsNativeFunctionWrapperHolder
+{
+    void *callbackState;
```

Comment:
There's also xplat compile errors claiming that these two values need a `Field` barrier, since they are in recycler memory. I think they both actually be a `FieldNoBarrier` since neither value is recycler allocated.

Reply:
@digitalinfinity I'd appreciate if you could double-check my thoughts here.

Reply:
Hmm I'll add `FieldNoBarrier`. 👍